src/nodes.js

```diff
@@ -79,13 +79,24 @@ var createNode = function(doc, nodeName, key, statics) {
 
 
 /**
+ * Properties for key maps.
+ */
+var keyMapProperties = {
+    isValid: {
+      value: true,
+      writable: true
+    }
+};
+
+
+/**
  * Creates a mapping that can be used to look up children using a key.
  * @param {!Element} el
  * @return {!Object<string, !Node>} A mapping of keys to the children of the
  *     Element.
  */
 var createKeyMap = function(el) {
-  var map = {};
+  var map = Object.create(null, keyMapProperties);
```

Comment:
IE9+. I'd rather see another property on the `NodeData`, since this'll also cause problems with the `isValid` key.